Added error mesh and material-not-found

AtlantiaKing opened the pull request:
Added rendering "ERR" mesh when a mesh is not loaded or not correctly loaded
Added pink material when a mesh is being rendered without a material

AtlantiaKing committed:
Added error mesh and MaterialNotFound material

A mesh that spells the letters "ERR" is now rendered for a mesh that isn't loaded or isn't loaded correctly (changed logic in DirectXMeshRenderer)
Added error.obj (the "ERR" mesh that is exactly 1x1x1 in size)
Added DirectXDefaults singleton who keeps track of error materials and meshes, it doesn't create the material or mesh unless they are needed
Added reload call to the DirectXDefaults class in the DirectXEngine

AtlantiaKing committed:
Removed shadow rendering on a Pos3D shader

AtlantiaKing committed:
Fixed values set on a materials not staying after reload

Every bool, float or vector is now saved inside a DirectXMaterial so it can be reapplied when the material gets reload.
Previously all the set information would be lost at Reload.

LeeVangraefschepe commented on Graphics/DirectX/DirectXDefaults.h:
```diff
@@ -0,0 +1,32 @@
+	class DirectXDefaults final : public Singleton<DirectXDefaults>
+	{
+	public:
+		virtual ~DirectXDefaults();
```
No rule of 5

LeeVangraefschepe commented on Graphics/DirectX/DirectXDefaults.h:
```diff
@@ -0,0 +1,32 @@
+		DirectXMaterial* GetMaterialError(ID3D11Device* pDevice);
+		void GetMeshError(ID3D11Device* pDevice, ID3D11Buffer*& pVertexBuffer, ID3D11Buffer*& pIndexBuffer, unsigned int& nrIndices);
+
+		void Reload(ID3D11Device* pDevice);
```
GetMeshError can be static
Reload can be const

LeeVangraefschepe commented on Graphics/DirectX/DirectXMeshRenderer.cpp:
```diff
@@ -41,18 +51,18 @@ void leap::graphics::DirectXMeshRenderer::Draw(IMaterial* pMaterial)
 	for (UINT p{}; p < techniqueDesc.Passes; ++p)
 	{
 		pDXMaterial->GetTechnique()->GetPassByIndex(p)->Apply(0, m_pDeviceContext);
-		m_pDeviceContext->DrawIndexed(m_NrIndices, 0, 0);
```
Result can be checked if it failed rendering.
pDXMaterial->GetTechnique()->GetDesc(&techniqueDesc);
pDXMaterial->GetTechnique()->GetPassByIndex(p)->Apply(0, m_pDeviceContext);

LeeVangraefschepe reviewed (approved):
Functionality wise is everything ok. But there are some code concerns. You got aproven but please look into the commented feedback.

AtlantiaKing committed:
Const correctness & static

Made GetMeshError method of the DirectXDefaults class a static method
Made Reload method of the DirectXDefaults class a const method

AtlantiaKing committed:
Added extra runtime exceptions to renderer

Added runtime exception when a meshrenderer fails to get the description from a effect technique
Added runtime exception when a meshrenderer fails to apply a effect technique pass
